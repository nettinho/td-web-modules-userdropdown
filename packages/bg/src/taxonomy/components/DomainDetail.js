import React from "react";
import PropTypes from "prop-types";
import { Icon, Header } from "semantic-ui-react";
import { connect } from "react-redux";
import { FormattedMessage } from "react-intl";
import { linkTo } from "../routes";
import DomainActions from "./DomainActions";

const toActionPath = linkTo.DOMAIN_ACTION;

export const DomainDetail = ({ type, name, description }) => {
  const available_actions = {
    update: {
      order: 2,
      icon: "edit",
      messageId: "actions.edit",
      action: id => toActionPath({ action: "edit", id })
    },
    deleteOption: true
  };
  return (
    <div style={{ marginTop: "1em" }}>
      <DomainActions availableActions={available_actions} />
      <Header as="h2">
        <Icon name="cube" />
        <Header.Content>
          {name}
          <Header.Subheader>
            {type || <FormattedMessage id="domain" />}
          </Header.Subheader>
        </Header.Content>
      </Header>
      <p>{description}</p>
    </div>
  );
};

DomainDetail.propTypes = {
  type: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string
};

const mapStateToProps = ({ domain }) => ({ ...domain });

export default connect(mapStateToProps)(DomainDetail);
