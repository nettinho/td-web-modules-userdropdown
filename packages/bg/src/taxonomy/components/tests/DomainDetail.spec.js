import React from "react";
import { shallow } from "enzyme";
import { DomainDetail } from "../DomainDetail";

describe("<DomainDetail />", () => {
  const props = { id: 1, type: "type", name: "name", description: "desc" };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<DomainDetail {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
