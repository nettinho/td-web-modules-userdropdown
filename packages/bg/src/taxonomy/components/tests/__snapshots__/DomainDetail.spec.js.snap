// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DomainDetail /> matches the latest snapshot 1`] = `
<div
  style={
    Object {
      "marginTop": "1em",
    }
  }
>
  <InjectIntl(Connect(DomainActions))
    availableActions={
      Object {
        "deleteOption": true,
        "update": Object {
          "action": [Function],
          "icon": "edit",
          "messageId": "actions.edit",
          "order": 2,
        },
      }
    }
  />
  <Header
    as="h2"
  >
    <Icon
      as="i"
      name="cube"
    />
    <HeaderContent>
      name
      <HeaderSubheader>
        type
      </HeaderSubheader>
    </HeaderContent>
  </Header>
  <p>
    desc
  </p>
</div>
`;
